onnx/defs/shape_inference.cc

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "shape_inference.h"
+
+namespace ONNX_NAMESPACE {
+
+void UnionShapeInfo(const TensorShapeProto& source_shape, TensorShapeProto& target_shape) {
```

Comment:
Thanks for the refactoring and moving it into a .cc file. May be useful to add a comment above that source_shape and target_shape are assumed to have same dim_size().

Reply:
I will remove it from the header